src/db.ts

```diff
@@ -31,28 +35,117 @@ import { DecoratedSchemaProcessor } from "./schema/decorated-schema-processor";
 import { Log } from "./utils/logger";
 import { DecoratorMetaStorage } from "./decorators/metadata/decorator-meta-storage";
 import { getDecoratorMetaStorage } from "./globals";
-import { CollectionNotFoundError } from "./error";
+import { CollectionNotFoundError, DatabaseBranchError } from "./error";
+import { Status } from "@grpc/grpc-js/build/src/constants";
 
-/**
- * Tigris Database
- */
 const SetCookie = "Set-Cookie";
 const Cookie = "Cookie";
 const BeginTransactionMethodName = "/tigrisdata.v1.Tigris/BeginTransaction";
 
+/**
+ * A branch name could be dynamically generated from environment variables.
+ *
+ * @example Simple name "my_database_branch" would translate to:
+ * ```
+ * {
+ *   name: "my_database_branch",
+ *   isTemplated: false
+ * }
+ * ```
+ * @example A dynamically generated branch name "my_db_${GIT_BRANCH}" would translate to:
+ * ```
+ * export GIT_BRANCH=feature_1
+ * {
+ *   name: "my_db_feature_1",
+ *   isTemplated: true
+ * }
+ * ```
+ */
+export type TemplatedBranchName = { name: string; isTemplated: boolean };
```

Comment:
do we need to call it `isTemplated` or something like `dynamicCreation: boolean`?